### PR TITLE
feat: refactor schema error handling from Go

### LINKED ISSUE #1299: It seems that errors.pb might persist across builds under some circumstances
I checked out online-boutique and was updating it to latest ftl. The first build failed because it was still using `ftl:verb` rather than `ftl:export`. I fixed that, but then the next build still resulted in the same errors:

```
trace: /xyz.block.ftl.v1.ControllerService/PullSchema (streaming client)
ftl: error: /Users/alec/dev/ftl-examples/online-boutique/backend/services/cart/cart.go:30:3-3: unexpected token "verb" (expecte
d Directive)
            /Users/alec/dev/ftl-examples/online-boutique/backend/services/cart/cart.go:43:3-3: unexpected token "verb" (expecte
d Directive)
            /Users/alec/dev/ftl-examples/online-boutique/backend/services/cart/cart.go:57:3-3: unexpected token "verb" (expecte
d Directive)
```

It seems like `errors.pb` is not being cleared between builds, resulting in previous errors persisting.

### PR DESCRIPTION
**Pull request by worstell**
- no longer pass Go compiler errors through the LSP
- instead of attempting to parse out schema/build errors from an aggregate error, we pass back schema errors and non-schema errors ("real" exceptions) explicitly and separately
- add errors build file to the ModuleConfig

fixes #1299

### CODE REVIEW

**Comment by wesbillman** [comment]
Screenshot 2024-04-18 at 5 09 32 PM
We might be getting duplicate errors now 🤔

File: buildengine/build.go [hunk 1]
```diff
@@ -33,6 +34,11 @@ func buildModule(ctx context.Context, sch *schema.Schema, module Module) error {
 	logger := log.FromContext(ctx).Scope(module.Module)
 	ctx = log.ContextWithLogger(ctx, logger)
 
+	// clear stale module errors before extracting schema
+	if err := os.RemoveAll(filepath.Join(module.AbsDeployDir(), module.Errors)); err != nil {
+		return fmt.Errorf("failed to clear errors: %w", err)
+	}
```
Comment (wesbillman) [inline buildengine/build.go:40]: Nice!

File: backend/schema/errors.go [hunk 2]
```diff
@@ -94,14 +94,15 @@ func Wrapf(pos Position, endColumn int, err error, format string, args ...any) *
 	return &e
 }
 
-func SortErrorsByPosition(merr []error) {
+func SortErrorsByPosition(merr []*Error) {
+	if len(merr) == 0 {
+		return
+	}
```
Comment (wesbillman) [inline backend/schema/errors.go:100]: Mostly just for my info, but is this check needed? Would just `sort.Slice` on an empty slice do the same thing?

**Comment by worstell** [comment]
> Screenshot 2024-04-18 at 5 09 32 PM We might be getting duplicate errors now 🤔

fixed!

**Review by wesbillman — APPROVED** [review]

**Comment by wesbillman** [comment]
Screenshot 2024-04-18 at 5 28 43 PM

It's beautiful!! :)

File: backend/schema/errors.go (hunk 2 above)
Comment (worstell) [reply backend/schema/errors.go]: nope you're totally right - removed!

**Pull request merged**